sentry-android-core/src/main/java/io/sentry/android/core/AndroidTransactionProfiler.java

```diff
@@ -230,13 +234,18 @@ public synchronized void onTransactionStart(@NotNull ITransaction transaction) {
       ProfilingTransactionData transactionData =
           transactionMap.get(transaction.getEventId().toString());
       if (transactionData != null) {
-        transactionData.notifyFinish(SystemClock.elapsedRealtimeNanos(), transactionStartNanos);
+        transactionData.notifyFinish(
+            SystemClock.elapsedRealtimeNanos(),
+            transactionStartNanos,
+            Process.getElapsedCpuTime(),
```

Comment:
@adinauer, I think this is what we also need to measure a transaction's CPU overhead. @stefanosiano, please correct me if I'm wrong. 

[SystemClock.elapsedRealtimeNanos()](https://developer.android.com/reference/android/os/SystemClock#elapsedRealtimeNanos())
> return the time since the system was booted, and include deep sleep. This clock is guaranteed to be monotonic, and continues to tick even when the CPU is in power saving modes, so is the recommend basis for general purpose interval timing.

[Process.getElapsedCpuTime()](https://developer.android.com/reference/android/os/Process#getElapsedCpuTime())

> Returns elapsed milliseconds of the time this process has run.

So you can calculate the CPU overhead by checking the duration of the transaction with `SystemClock.elapsedRealtimeNanos()` and comparing it with the milliseconds of the process has actually run `Process.getElapsedCpuTime()`.

How expensive are the API calls, @stefanosiano?

Reply:
The apis should be quick, even if i don't have any number to tell you.
I'm not sure on what you mean by cpu overhead. I mean, if a transaction lasts longer, both system clock and cpu time increase.
The difference between them should be `SystemClock.elapsedRealtimeNanos()` starts since system boot, while `Process.getElapsedCpuTime()` since the process start. But during a transaction they should increase in the same way (except for deep sleep, but that's not important here now, imho).
Am i missing something?

Reply:
Hmm, maybe I still don't get the actual difference between [SystemClock.elapsedRealtimeNanos()](https://developer.android.com/reference/android/os/SystemClock#elapsedRealtimeNanos()) and [Process.getElapsedCpuTime()](https://developer.android.com/reference/android/os/Process#getElapsedCpuTime()). Nevermind, @adinauer is looking into this right now and he will know.